src/sentry/api/helpers/group_index/index.py

```diff
@@ -89,31 +90,46 @@ def build_query_params_from_request(
     query = query.strip()
 
     if request.GET.get("savedSearch") == "0" and request.user and not has_query:
-        saved_searches = (
-            SavedSearch.objects
-            # Do not include pinned or personal searches from other users in
-            # the same organization. DOES include the requesting users pinned
-            # search
-            .exclude(
-                ~Q(owner_id=request.user.id),
-                visibility__in=(Visibility.OWNER, Visibility.OWNER_PINNED),
-            )
-            .filter(
-                Q(organization=organization) | Q(is_global=True),
-            )
-            .extra(order_by=["name"])
-        )
-        selected_search_id = request.GET.get("searchId", None)
-        if selected_search_id:
-            # saved search requested by the id
-            saved_search = saved_searches.filter(id=int(selected_search_id)).first()
-        else:
-            # pinned saved search
-            saved_search = saved_searches.filter(visibility=Visibility.OWNER_PINNED).first()
+        if features.has("organizations:issue-stream-custom-views", organization):
+            selected_view_id = request.GET.get("groupSearchViewId")
+            if selected_view_id:
+                default_view = GroupSearchView.objects.get(id=int(selected_view_id))
+            else:
+                default_view = GroupSearchView.objects.get(
```

Comment:
Does this cover the case where the user has no default views (and we should give them the prioritized tab)?

Reply:
oh and I also think `.get()` will error here if it doesn't find anything which we don't want

Reply:
I think the `.get()` is *technically fine since this part of the code should not be reachable if the user has no custom views, but that would have to be enforced by the frontend, so I think, to your point, it's probably still better to double check here as well.

Reply:
Maybe I'm forgetting how this works, but don't users start with no views? I thought that we only created views when they modified them.

Reply:
@malwilley okay so after confirming with Richard I think my previous understanding of how this logic worked was incorrect - feel free to just ignore everything I just said above. I believe the most up to date version addresses your concerns and is the correct implementation now.